test/organization_test.go

```diff
@@ -122,7 +132,20 @@ modfor:
 	}
 
 	t.Log("update")
-	org.Labels = map[string]string{"aa": "bb"}
+	updateCtx, cancel := context.WithTimeout(ctx, 30*time.Second)
+	t.Cleanup(cancel)
+	require.NoError(t, wait.PollUntil(time.Second, func() (done bool, err error) {
+		if err := cl.Get(ctx, types.NamespacedName{Name: "test"}, org); err != nil {
+			t.Log("getting organization", err)
+			return false, nil
+		}
+		org.Labels = map[string]string{"aa": "bb"}
+		if err := cl.Update(ctx, org); err != nil {
```

Comment:
only retry if err modified, otherwise return the error early

Reply:
I don't get it.
If the resource version is out dated for few times, why do we return error early instead of retrying?